### PR TITLE
[Refactor] check status of functions

### PR DESCRIPTION
**Pull request by packy92**
Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 0 / 3 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/aggregator.h|0|2|00.00%|[360, 385]|
|:large_blue_circle:|src/exec/pipeline/aggregate/aggregate_distinct_blocking_sink_operator.h|0|1|00.00%|[71]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 2 / 6 (33.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/aggregator.h|0|3|00.00%|[360, 371, 385]|
|:large_blue_circle:|src/exec/pipeline/aggregate/aggregate_distinct_blocking_sink_operator.h|0|1|00.00%|[71]|
|:large_blue_circle:|src/exec/stream/aggregate/agg_group_state.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/exec/stream/aggregate/agg_state_data.h|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 2 / 6 (33.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/aggregator.h|0|3|00.00%|[360, 371, 385]|
|:large_blue_circle:|src/exec/pipeline/aggregate/aggregate_distinct_blocking_sink_operator.h|0|1|00.00%|[71]|
|:large_blue_circle:|src/exec/stream/aggregate/agg_group_state.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/exec/stream/aggregate/agg_state_data.h|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 2 / 6 (33.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/aggregator.h|0|3|00.00%|[360, 371, 385]|
|:large_blue_circle:|src/exec/pipeline/aggregate/aggregate_distinct_blocking_sink_operator.h|0|1|00.00%|[71]|
|:large_blue_circle:|src/exec/stream/aggregate/agg_group_state.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/exec/stream/aggregate/agg_state_data.h|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/pipeline/aggregate/aggregate_distinct_blocking_sink_operator.h|0|1|00.00%|[71]|
|:large_blue_circle:|src/exec/stream/aggregate/agg_group_state.cpp|1|1|100.00%|[]|

**Review by liuyehcf — DISMISSED** [review]

File: be/src/exec/pipeline/aggregate/spillable_aggregate_distinct_blocking_operator.h [hunk 1]
```diff
@@ -137,7 +137,7 @@ class SpillableAggregateDistinctBlockingSourceOperatorFactory final : public Sou
 
     ~SpillableAggregateDistinctBlockingSourceOperatorFactory() override = default;
 
-    Status prepare(RuntimeState* state) override;
+    [[nodiscard]] Status prepare(RuntimeState* state) override;
 
     OperatorPtr create(int32_t degree_of_parallelism, int32_t driver_sequence) override;
```
[the remark this answers is not in the document]
Comment (kangkaisen) [reply be/src/exec/pipeline/aggregate/spillable_aggregate_distinct_blocking_operator.h:143]: Duplicate Private Keyword: It appears to be an unintentional error having private written twice consecutively without any member declarations between.

File: be/src/exec/stream/aggregate/agg_group_state.h [hunk 2]
```diff
@@ -32,31 +32,31 @@ class AggGroupState {
...
 
 private:
-    Status _prepare_mem_state_tables(RuntimeState* state, const std::vector<AggStateData*>& intermediate_agg_states,
+    [[discard]] Status _prepare_mem_state_tables(RuntimeState* state, const std::vector<AggStateData*>& intermediate_agg_states,
                                      const std::vector<AggStateData*>& detail_agg_states);
-    Status _prepare_imt_state_tables(RuntimeState* state, const std::vector<AggStateData*>& intermediate_agg_states,
+    [[discard]] Status _prepare_imt_state_tables(RuntimeState* state, const std::vector<AggStateData*>& intermediate_agg_states,
                                      const std::vector<AggStateData*>& detail_agg_states);
     StateTable* _find_detail_state_table(const AggStateDataUPtr& agg_state) const;
     ChunkPtr _build_intermediate_chunk(const Columns& group_by_columns, const Columns& agg_intermediate_columns) const;
```
[the remark this answers is not in the document]
Comment (kangkaisen) [reply be/src/exec/stream/aggregate/agg_group_state.h:67]: Function parameter const correctness: If you have parameters that shouldn't be modified by the function they're passed into, mark them as const.

File: be/src/exec/stream/aggregate/stream_aggregator.h [hunk 3]
```diff
@@ -57,51 +57,51 @@ class StreamAggregator final : public Aggregator {
...
-    Status _output_final_result_with_retract(size_t chunk_size, const Columns& group_by_columns,
+    [[discard]] Status _output_final_result_with_retract(size_t chunk_size, const Columns& group_by_columns,
                                              ChunkPtr* post_chunk_result);
 
     // Output results without generating retract messages.
-    Status _output_result_changes_without_retract(size_t chunk_size, const Columns& group_by_columns,
+    [[discard]] Status _output_result_changes_without_retract(size_t chunk_size, const Columns& group_by_columns,
                                                   StreamChunkPtr* result_chunk);
 
 private:
```
[the remark this answers is not in the document]
Comment (kangkaisen) [reply be/src/exec/stream/aggregate/stream_aggregator.h:108]: Function overloading: output_changes function is overloaded twice. If the difference between them is significant, consider different function names.

File: be/src/exec/stream/aggregate/agg_group_state.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (packy92) [reply be/src/exec/stream/aggregate/agg_group_state.h:67]: we haven't implemented `_prepare_imt_state_tables`. `state` may be modified.

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/pipeline/aggregate/aggregate_distinct_blocking_sink_operator.h|0|1|00.00%|[71]|
|:large_blue_circle:|src/exec/stream/aggregate/agg_group_state.cpp|1|1|100.00%|[]|

**Review by liuyehcf — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 3 / 4 (75.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/pipeline/aggregate/aggregate_distinct_blocking_sink_operator.h|0|1|00.00%|[71]|
|:large_blue_circle:|src/exec/stream/aggregate/stream_aggregator.cpp|2|2|100.00%|[]|
|:large_blue_circle:|src/exec/stream/aggregate/agg_group_state.cpp|1|1|100.00%|[]|

**Pull request merged**